### ISSUE TITLE
Opening the menu will glitch Kraid's graphics

### ISSUE THREAD

**Issue by MarioFanGamer** [tewtal/sm_practice_hack#63]
The title mentions it all. In fact, I even know the reason as well as exactly why it happens: The menu is on layer 3. Naturally, it causes layer 3 in the gameplay to temporarily disappear as well as Ceres's first room to glitch. This fine for most part since the effect is only temporal. The issue here is that it _isn't_ for Kraid where opening the menu will glitch him permanently (-ish since unpausing will fix the tilemap for the most part). I even know the reason since I have checked the game in a debugger: He reallocates the BG2 tilemap to $4000, where usually BG3 charset is located, and moves the latter to $2000 ($8000 and $4000 in debuggers, respectively) while the menu always assumes the charset doesn't change.

It is a noteworthy bug to be fixed IMO because opening the menu in Kraid's battle [can realistically happen and hinders the gameplay](https://youtu.be/qg3_yrHIAzk?t=680).

**Comment by idlechild** [comment]
Yep it's a known issue; it also affects the minimap if you have that turned on (even when Kraid is already dead).  I was trying to fix just that issue and didn't have much luck... but I didn't know where the BG3 charset was being relocated to.  With that information it might be more reasonable to fix this issue.

**Comment by idlechild** [comment]
I fixed the glitched Kraid graphics, hopefully without breaking anything else
https://github.com/tewtal/sm_practice_hack/pull/48/commits/cb9e832c1a283421c9dab897fe954293a0c9973c
When in Kraid's room, I transfer VRAM to 0x2000 instead of 0x4000, and I also had to update register $210C where we tell the hardware the starting location of BG3.
